File: 6.js
class Stack {
  /**
   * Stack Obyekt
   */
  constructor() {
    this.stack = [];
  }

  /**
   * bo'sh ekanligini tekshirish
   */
  isEmpty() {
    return this.stack.length == 0;
  }

  /**
   * Element qo'shish,
   * @param newData yangi elementni kiriting
   */
  add(newData) {
    this.stack.push(newData);
    return true;
  }

  /**
   * Elementni sug'urib olish
   */
  pop() {
    if (this.isEmpty()) {
      return "Stack is empty";
    } else {
      return this.stack.pop();
    }
  }
  /**
   * Eng ustki elementni ko'rish
   */
  peek() {
    if (this.isEmpty()) {
      return "Stack is empty";
    } else {
      let l = this.stack.length;
      return this.stack[l - 1];
    }
  }
}
const stack = new Stack();
stack.add(5);
stack.add(6);
stack.add(7);

console.log("Top element: ", stack.peek())
console.log(stack.pop())
console.log(stack.pop())
console.log(stack.pop())
console.log(stack.pop())

File: 6.py
# theme: Stack Data Structure
class Stack:
    """Stack Obyekti"""
    def __init__(self):
        self.stack = []
    
    def isEmpty(self):
        """Ro'yxat bo'sh ekanligini tekshirish"""
        return len(self.stack) == 0;
    
    def push(self,newData):
        """Element qo'shish"""
        self.stack.append(newData)
        return True;
    
    def pop(self):
        """Element sug'urib olish"""
        if self.isEmpty():
            return "Stack is empty"
        else:
            return self.stack.pop()
    
    def peek(self):
        """Eng ustki elementni ko'rish"""
        if self.isEmpty():
            return "Stack is empty"
        else:
            return self.stack[-1]


if __name__=='__main__':
    stack = Stack()
    stack.push(5)
    stack.push(6)
    stack.push(7)
    print("Top element: ", stack.peek())
    print(stack.pop())
    print(stack.pop())
    print(stack.pop())
    print(stack.pop())